inc/author-structure-data/class-vk-author-structure-data.php

```diff
@@ -0,0 +1,201 @@
+
+		?>
+	<h2>構造化データ</h2>
+	<table class="form-table">
+	  <tr>
+		<th><label for='author_type'>@type</label></th>
+		<td>
+		  <select name='author_type' id='author_type'>
+		  <option value='organization' <?php echo ( ( $author_type == 'organization' ) ) ? 'selected' : ''; ?> >Organization</option>
+		  <option value='person' <?php echo ( ( $author_type == 'person' ) ) ? 'selected' : ''; ?> >Person</option>
+		  </select>
+		  <p class="discription">著者が個人のときは Person を、著者が組織のときは Organization を選択してください。</p>
+		</td>
+	  </tr>
+	  <tr>
+		<th><label for='author_name'>name</label></th>
+		<td>
+		  <label><input id='author_name' type='text' name='author_name' value='<?php echo esc_attr( $author_name ); ?>'/></label>
+		  <p class="discription">未入力の場合、ブログ上の表示名が使用されます。</p>
+		</td>
+	  </tr>
+	  <tr>
+		<th><label for='author_url'>url</label></th>
+		<td>
+		  <label><input id='author_url' type='url' name='author_url' value='<?php echo esc_attr( $author_url ); ?>'/></label>
+		  <p class="discription">
+			このユーザーのプロフィールページのURLを入力してください。
+			未入力の場合
+			@type が 個人の場合 : このサイトの投稿者アーカイブページのURLが使用されます。
+			@type が 組織の場合 : このホームページのトップページのURLが適用されます。
+			※ ユーザープロフィールの連絡先情報に指定したサイトのURLは、url には反映されません
+		  </p>
+		</td>
+	  </tr>
+	  <tr>
+		<th><label for='author_sameAs'>sameAs</label></th>
+		<td>
+		  <label><input id='author_sameAs' type='url' name='author_sameAs' value='<?php echo esc_attr( $author_sameAs ); ?>'/></label>
+		  <p class="discription">SNSやWikipediaなどのプロフィールURLを指定します。</p>
+		</td>
+	  </tr>
+	</table>
+		<?php
+	}
+
+	/**
+	 * Update Author Structure Date
+	 */
+	public static function update_structure_data( $user_id, $old_user_data ) {
+		if ( isset( $_POST['author_type'] ) ) {
+			update_user_meta( $user_id, 'author_type', $_POST['author_type'], $old_user_data->author_type );
+		}
+		if ( isset( $_POST['author_name'] ) ) {
+			update_user_meta( $user_id, 'author_name', $_POST['author_name'], $old_user_data->author_name );
+		}
+		if ( isset( $_POST['author_url'] ) ) {
+			update_user_meta( $user_id, 'author_url', $_POST['author_url'], $old_user_data->author_url );
+		}
+		if ( isset( $_POST['author_sameAs'] ) ) {
+			update_user_meta( $user_id, 'author_sameAs', $_POST['author_sameAs'], $old_user_data->author_sameAs );
+		}
+	}
+
+	/**
+	 * json-LD
+	 */
+	public static function print_jsonLD_in_head() {
+		global $post;
+		$author_id = $post->post_author;
+		if ( is_single() ) {
+			$author_array = self::generate_author_array( $author_id );
+			if ( $author_array && is_array( $author_array ) ) {
+				echo '<!-- [ VK All in One Expansion Unit Structure Data ] -->';
+				echo '<script type="application/ld+json">' . json_encode( $author_array, JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE ) . '</script>';
+				echo '<!-- [ / VK All in One Expansion Unit Structure Data ] -->';
+			}
+		}
+	}
+
+	/**
+	 * ユーザー情報を配列で返す
+	 *
+	 * @return array $author_array
+	 */
+	public static function generate_author_array( $author_id = '' ) {
```

Comment:
@doshimaf 
値を返す系は get_
エスケープした上で echo する系は the_
で統一したいっす。
```suggestion
	public static function get_generate_author_array( $author_id = '' ) {
```